pointmatcher/Registrar.h

```diff
@@ -90,80 +90,79 @@ namespace PointMatcherSupport
 			//! Return the available parameters for this class
 			virtual const Parametrizable::ParametersDoc availableParameters() const = 0;
 		};
-		
+
 		//! A descriptor for a class C that provides parameters
 		template<typename C>
 		struct GenericClassDescriptor: public ClassDescriptor
 		{
 			virtual std::shared_ptr<Interface> createInstance(const std::string& className, const Parametrizable::Parameters& params) const
 			{
 				std::shared_ptr<C> instance = std::make_shared<C>(params);
-				
+
 				// check that all parameters were set
-				for (BOOST_AUTO(it, params.begin()); it != params.end() ;++it)
+				for (const auto& param : params)
 				{
-					if (instance->parametersUsed.find(it->first) == instance->parametersUsed.end()){
+					if (instance->parametersUsed.find(param.first) == instance->parametersUsed.end()){
 						throw Parametrizable::InvalidParameter(
-							(boost::format("Parameter %1% for module %2% was set but is not used") % it->first % className).str()
+							(boost::format("Parameter %1% for module %2% was set but is not used") % param.first % className).str()
 						);
 					}
 				}
-				
 				return instance;
 			}
+
 			virtual const std::string description() const
 			{
 				return C::description();
 			}
+
 			virtual const Parametrizable::ParametersDoc availableParameters() const
 			{
 				return C::availableParameters();
 			}
 		};
-		
+
 		//! A descriptor for a class C that does not provide any parameter
 		template<typename C>
 		struct GenericClassDescriptorNoParam: public ClassDescriptor
 		{
 			virtual std::shared_ptr<Interface> createInstance(const std::string& className, const Parametrizable::Parameters& params) const
 			{
-				for (BOOST_AUTO(it, params.begin()); it != params.end() ;++it)
+				for (const auto& param : params)
 					throw Parametrizable::InvalidParameter(
-							(boost::format("Parameter %1% was set but module %2% dos not use any parameter") % it->first % className).str()
+							(boost::format("Parameter %1% was set but module %2% dos not use any parameter") % param.first % className).str()
 						);
+
 				return std::make_shared<C>();
 			}
+
 			virtual const std::string description() const
 			{
 				return C::description();
 			}
+
 			virtual const Parametrizable::ParametersDoc availableParameters() const
 			{
 				return Parametrizable::ParametersDoc();
 			}
 		};
-		
+
 	protected:
-		typedef std::map<std::string, ClassDescriptor*> DescriptorMap; //!< descriptors for sub-classes of Interface, indexed by their names
+		typedef std::map<std::string, std::shared_ptr<ClassDescriptor>> DescriptorMap; //!< descriptors for sub-classes of Interface, indexed by their names
```

Comment:
Does it make sense to use `using` here?

https://www.internalpointers.com/post/differences-between-using-and-typedef-modern-c

Reply:
Hi @YoshuaNava,
I'm sorry, but I'm not quite sure to understand what you mean here. Do you mean creating an alias for `std::shared_ptr<ClassDescriptor>` or replacing the `typedef` syntax for the `using` syntax?

@pomerlef and @simonpierredeschenes what do you think?

Reply:
In here I mean replacing typedef with using.

In the later comments I do propose an alias for the shared pointer of the class descriptor.

Reply:
To be frank, I never used `using` in C++ except in `using namespace ...`, but I'm not an expert neither. @YoshuaNava is most probably in a better position to answer that.

When I'm search for `using` in the library, it's clearly not uniform. A recent contribution in octree is using it, but that's about it. To not branch multiple PR in one, I would keep `typedef` here and open an issue about using `using` properly through the code.

Reply:
@pomerlef Thanks. Agreed 👍